src/nrniv/nrncore_write/data/cell_group.cpp

```diff
@@ -110,7 +116,15 @@ CellGroup* CellGroup::mk_cellgroups(CellGroup* cgs) {
                     } else { // if an acell is never a source, it will not have a presyn
                         cgs[i].output_gid[npre] = -1;
                     }
-                    // the way we associate an acell PreSyn with the Point_process.
+                    // the way we associate an acell PreSyn with the
+                    // Point_process.
+                    if (agid < std::numeric_limits<int>::min() ||
+                        agid > std::numeric_limits<int>::max()) {
+                        std::ostringstream oss;
+                        oss << "cannot store cgs[" << i << "].output_vindex[" << npre
+                            << "]=" << agid;
```

Comment:
`nitpick`: let's change error message to something like: `integer overflow : maximum X artificial cells can be created on a single thread`.